### ISSUE TITLE
[WIN] UDS messageInterpreter (Rx)

### ISSUE THREAD

**Issue by Nahrain1** [amosproj/amos2024ss07-updating-flash-boot-loader#41]
## User story
1. As a developer
2. I need a the UDS class to check what kind of message it is
3. So that the received messages are passed along to the correct module

## Acceptance criteria
* Messages are received from Communication Layer
* Parse response to Getting System Name (0xF197) of Read Data By Identifier (SID 0x22)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
